### PR TITLE
Add smoke tests cadence values / storage operations

### LINKED ISSUE #1178: Add smoke tests cadence values / storage operations
### Issue To Be Solved

Add smoke tests to cover the new cadence values backed by the new storage.
- Test storage operations (container insert, remove, update, etc.)
- Cover all values. More focus on arrays, dictionaries, and composites 
- Include nested values

### PR DESCRIPTION
**Pull request by SupunS**
Closes #1178

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/tests/interpreter/values_test.go [hunk 1]
```diff
@@ -0,0 +1,1547 @@
...
+			orgValue, ok := entries.get(key)
+			require.True(t, ok, "cannot fine key: %v", key)
+
+			utils.AssertValuesEqual(t, inter, orgValue, value)
+			return true
+		})
+	})
+
+	t.Run("deep copy", func(t *testing.T) {
+		newOwner := atree.Address([8]byte{'B'})
```
Comment (turbolent) [inline runtime/tests/interpreter/values_test.go:122]:
```suggestion
		newOwner := atree.Address{'B'}
```

File: runtime/tests/interpreter/values_test.go [hunk 2]
```diff
@@ -0,0 +1,1547 @@
...
+			elementByIndex := testArray.Get(inter, interpreter.ReturnEmptyLocationRange, index)
+			utils.AssertValuesEqual(t, inter, element, elementByIndex)
+
+			index++
+			return true
+		})
+	})
+
+	t.Run("deep copy", func(t *testing.T) {
+		newOwner := atree.Address([8]byte{'B'})
```
Comment (turbolent) [inline runtime/tests/interpreter/values_test.go:491]:
```suggestion
		newOwner := atree.Address{'B'}
```

File: runtime/tests/interpreter/values_test.go [hunk 3]
```diff
@@ -0,0 +1,1547 @@
...
+			require.True(t, ok)
+			utils.AssertValuesEqual(t, inter, orgValue, value)
+			fieldCount++
+		})
+
+		assert.Equal(t, len(orgFields), fieldCount)
+	})
+
+	t.Run("deep copy", func(t *testing.T) {
+		newOwner := atree.Address([8]byte{'B'})
```
Comment (turbolent) [inline runtime/tests/interpreter/values_test.go:842]:
```suggestion
		newOwner := atree.Address{'B'}
```

File: runtime/tests/interpreter/values_test.go [hunk 4]
```diff
@@ -0,0 +1,1547 @@
...
+			value := testComposite.GetField(inter, interpreter.ReturnEmptyLocationRange, name)
+			utils.AssertValuesEqual(t, inter, orgValue, value)
+		}
+
+		owner := testComposite.GetOwner()
+		assert.Equal(t, orgOwner[:], owner[:])
+	})
+
+	t.Run("remove field", func(t *testing.T) {
+		newOwner := atree.Address([8]byte{'c'})
```
Comment (turbolent) [inline runtime/tests/interpreter/values_test.go:882]:
```suggestion
		newOwner := atree.Address{'c'}
```

File: runtime/tests/interpreter/values_test.go [hunk 5]
```diff
@@ -0,0 +1,1547 @@
...
+
+			value, found := testMap.Get(inter, interpreter.ReturnEmptyLocationRange, orgKey)
+			require.True(t, found)
+			utils.AssertValuesEqual(t, inter, orgValue, value)
+
+			return false
+		})
+
+		owner := testMap.GetOwner()
+		assert.Equal(t, orgOwner[:], owner[:])
```
Comment (turbolent) [inline runtime/tests/interpreter/values_test.go:106]: Is it possible to compare the values directly or is the slicing necessary?

**Review by turbolent — APPROVED** [review]
Nice work!

File: runtime/tests/interpreter/values_test.go (hunk 1 above)
Comment (SupunS) [reply runtime/tests/interpreter/values_test.go:122]: oh, I'm not sure how this came back 🤔   I refactored this earlier: ee6e20762d472afb882c9ac677700e2a20e326e5

Thanks for catching it! Will clean it up.

File: runtime/tests/interpreter/values_test.go (hunk 5 above)
Comment (SupunS) [reply runtime/tests/interpreter/values_test.go:106]: Yes, should be possible to compare directly

**Review by fxamacker — COMMENTED** [review]
Not sure, but do we want to call `atree.CheckStorageHealth` in smoke tests?

File: runtime/tests/interpreter/values_test.go [hunk 6]
```diff
@@ -0,0 +1,1565 @@
...
+
+	t.Run("deep copy", func(t *testing.T) {
+		newOwner := atree.Address{'B'}
+		copyOfTestMap = testMap.Transfer(
+			inter,
+			interpreter.ReturnEmptyLocationRange,
+			newOwner,
+			false,
+			nil,
+		).(*interpreter.DictionaryValue)
```
Comment (fxamacker) [inline runtime/tests/interpreter/values_test.go:147]: Do we need to smoke test `Transfer()` with `remove` flag set to `true`?  Same question for `ArrayValue` and `CompositeValue`.

**Review by fxamacker — APPROVED** [review]
Great work!  :+1: 

Just one comment.

**Comment by SupunS** [comment]
> Not sure, but do we want to call atree.CheckStorageHealth in smoke tests?

@fxamacker I had added this earlier, but it was taking ages to run the test with the validation. So removed it eventually. It's good to have it too, maybe by reducing the value count (we would have to compromise one)?

**Comment by fxamacker** [comment]
> > Not sure, but do we want to call atree.CheckStorageHealth in smoke tests?
> 
> @fxamacker I had added this earlier, but it was taking ages to run the test with the validation. So removed it eventually. It's good to have it too, maybe by reducing the value count (we would have to compromise one)?

Oh, I see.  I think we can lower the value count.  I think it'll take ~1500 boolean values to cause slab split but most other types will require fewer.  

Maybe we can also run storage health check selectively, such as only in "random insert & remove"?  If it still takes too long, we can add command-line flag to enable this check explicitly.

File: runtime/tests/interpreter/values_test.go (hunk 6 above)
Comment (SupunS) [reply runtime/tests/interpreter/values_test.go:150]: Added in a6d8aa1a4a09979404f17a4c8e103b461be4075a

**Pull request merged**